pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1072,36 +1127,52 @@ func (n *NetworkPolicyController) syncAddressGroup(key string) error {
 	groupSelector := addressGroup.Selector
 	if groupSelector.Namespace != "" {
 		// Namespace presence indicates Pods must be selected from the same Namespace.
-		pods, _ = n.podLister.Pods(groupSelector.Namespace).List(groupSelector.PodSelector)
-	} else if groupSelector.NamespaceSelector != nil && groupSelector.PodSelector != nil {
-		// Pods must be selected from Namespaces matching nsSelector.
+		if groupSelector.PodSelector != nil {
```

Comment:
why should PodSelector and ExternalEntitySelector being mutually exclusive? We can have a policy refers both Pods and ExternalEntity(VM etc), no?

Reply:
Fixed, thanks for the catch.